libraries/botbuilder-dialogs-adaptive/src/actions/continueConversationLater.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    BoolExpression,
+    BoolExpressionConverter,
+    Expression,
+    StringExpression,
+    StringExpressionConverter,
+    ValueExpression,
+    ValueExpressionConverter,
+} from 'adaptive-expressions';
+import { Activity, ActivityEventNames, ActivityTypes, QueueStorage, TurnContext } from 'botbuilder-core';
+import {
+    Converter,
+    ConverterFactory,
+    Dialog,
+    DialogConfiguration,
+    DialogContext,
+    DialogTurnResult,
+    DialogTurnStateConstants,
+} from 'botbuilder-dialogs';
+
+export interface ContinueConversationLaterConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | Expression | BoolExpression;
+    date?: string | Expression | StringExpression;
+    value?: unknown | ValueExpression;
+}
+
+/**
+ * Action which schedules a conversation to be continued later by writing an EventActivity(Name=ContinueConversation) to a queue.
+ */
+export class ContinueConversationLater<O extends object = {}>
+    extends Dialog<O>
+    implements ContinueConversationLaterConfiguration {
+    public static $kind = 'Microsoft.ContinueConversationLater';
+
+    /**
+     * Gets or sets an optional expression which if is true will disable this action.
+     */
+    public disabled: BoolExpression;
+
+    /**
+     * Gets or sets the expression which resolves to the date/time to continue the conversation.
+     */
+    public date: StringExpression;
+
+    /**
+     * Gets or sets the value to pass in the EventActivity payload.
+     */
+    public value: ValueExpression;
+
+    public getConverter(property: keyof ContinueConversationLaterConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'date':
+                return new StringExpressionConverter();
+            case 'value':
+                return new ValueExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the [Dialog](xref:botbuilder-dialogs.Dialog) is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional. Initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A `Promise` representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: O): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled.getValue(dc.state)) {
+            return await dc.endDialog();
```

Comment:
Shouldn't need to use the `await` keyword here since you're already returning a promise.